### ISSUE TITLE
`ttnn.max_pool2d` returns incorrect results with stride (1, 1)

### ISSUE THREAD

**Issue by jerrysky3** [tenstorrent/tt-metal#14260]
**Describe the bug**

When `ttnn.max_pool2d` runs with kernel `(3, 3)`, stride `(1, 1)`, and padding `(1, 1)`, it produces incorrect results at the last row.

Here is the output I got from an all 1. input tensor:

```
TorchTensor([[[[1.0000e+00, 1.0000e+00, 1.0000e+00,  ..., 1.0000e+00,
                1.0000e+00, 1.0000e+00],
               [1.0000e+00, 1.0000e+00, 1.0000e+00,  ..., 1.0000e+00,
                1.0000e+00, 1.0000e+00],
               [1.0000e+00, 1.0000e+00, 1.0000e+00,  ..., 1.0000e+00,
                1.0000e+00, 1.0000e+00],
               ...,
               [1.0000e+00, 1.0000e+00, 1.0000e+00,  ..., 1.0000e+00,
                1.0000e+00, 1.0000e+00],
               [1.0000e+00, 1.0000e+00, 1.0000e+00,  ..., 1.0000e+00,
                1.0000e+00, 1.0000e+00],
               [1.0000e+00, 1.0000e+00, 1.0000e+00,  ..., 1.0000e+00,
                3.3895e+38,        inf]]]], dtype=torch.bfloat16) torch.Size([1, 1, 81, 32])
```

There are some unexpected random numbers in the last row

**To Reproduce**

Run the example below (I tested it on n300):

```py
import torch
import ttnn

def main(device):
    input = torch.ones((1, 32, 9, 9), dtype=torch.bfloat16)
    expected = torch.nn.functional.max_pool2d(input, (3, 3), stride=1, padding=1, dilation=1)
    input = input.reshape((input.shape[0], 1, input.shape[1], input.shape[2] * input.shape[3])).permute((0, 1, 3, 2))

    input_tensor = ttnn.from_torch(input, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
    input_tensor = ttnn.to_memory_config(input_tensor, ttnn.L1_MEMORY_CONFIG)
    result = ttnn.max_pool2d(input_tensor, 1, 9, 9, 32, [3, 3], [1, 1], [1, 1], [1, 1])
    result = ttnn.to_torch(result)
    print(result, result.shape)

    expected = expected.reshape(
        (expected.shape[0], 1, expected.shape[1], expected.shape[2] * expected.shape[3])
    ).permute((0, 1, 3, 2))
    print(torch.allclose(result, expected))

if __name__ == "__main__":
    device = ttnn.open_device(device_id=0, l1_small_size=65536)
    try:
        main(device)
    finally:
        ttnn.close_device(device)
```

**Expected behavior**

ttnn output should be all 1.0 and match pytorch output

**Comment by mywoodstock** [comment]
Bugfix PR: https://github.com/tenstorrent/tt-metal/pull/14307
